lib/src/main_common.dart

```diff
@@ -3,13 +3,16 @@ import 'package:allthenews/src/app/app_flavors.dart';
 import 'package:allthenews/src/di/injector.dart';
 import 'package:allthenews/src/ui/common/theme/theme_notifier.dart';
 import 'package:allthenews/src/ui/pages/presentation/presentation_notifier.dart';
+import 'package:firebase_core/firebase_core.dart';
 import 'package:firebase_crashlytics/firebase_crashlytics.dart';
 import 'package:flutter/material.dart';
 import 'package:provider/provider.dart';
 
-void mainCommon(Environment flavor) {
+Future<void> mainCommon(Environment flavor) async {
   injectDependencies(flavor);
-  FlutterError.onError = Crashlytics.instance.recordFlutterError;
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp();
```

Comment:
await w mainie blokuje wątek więc trochę słabo - i widzę, że tak jest w example na pub.dev, ale w dokumentacji już dużo ładniej to ograli: 
https://firebase.flutter.dev/docs/overview#initializing-flutterfire
Generalnie jest issue, że na flutterze 1.20 nie śmiga, ale fixem jest pozostawienie `WidgetsFlutterBinding.ensureInitialized()` w mainie :) 
Tu wspomniane issue: https://github.com/FirebaseExtended/flutterfire/issues/3490

Reply:
miałem to na liście rzeczy do poprawienia, ale cocoapods na iosie się zjebał i potem już zapomniałem :p